web/src/composebox_typeahead.ts

```diff
@@ -996,6 +999,57 @@ export function content_highlighter_html(item: TypeaheadSuggestion): string | un
     }
 }
 
+function will_produce_broken_stream_topic_link(word: string): boolean {
+    return invalid_stream_topic_regex.test(word);
+}
+
+function get_stream_name_from_topic_link_syntax(syntax: string): string {
+    const start = syntax.lastIndexOf("#**");
+    const end = syntax.lastIndexOf(">");
+    return syntax.slice(start + 3, end);
+}
+
+export function html_escape_markdown_syntax_characters(text: string): string {
+    return text.replaceAll(invalid_stream_topic_regex, (match) => {
+        switch (match) {
+            case "`":
+                return "&grave;";
+            case ">":
+                return "&gt;";
+            case "*":
+                return "&#42;";
+            case "$$":
+                return "&#36;&#36;";
+            default:
+                return match;
+        }
+    });
+}
+
+function url_syntax(stream_name: string, topic_name: string | undefined): string {
```

Comment:
Why is this called `url_syntax`? A function in the typeahead with this name is going to be assumed to be about something totally different.

Maybe it would also be a good idea to put all these new functions in a new `topic_link_util.ts` module?

Then this could be something like `topic_link_util.get_fallback_markdown_link()`

Reply:
Putting these in a module would be a good idea, since these functions are also required in #29302